src/main/java/frc/robot/subsystems/conveyer/Conveyer.java

```diff
@@ -0,0 +1,144 @@
+
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setDegreesFromGearRatio(frontMotorEncoder, ConveyerConstants.FRONT_GEAR_RATIO));
+    errors +=
+        SparkMaxUtils.check(
+            SparkMaxUtils.UnitConversions.setDegreesFromGearRatio(backMotorEncoder, ConveyerConstants.BACK_GEAR_RATIO));
+
+    errors += SparkMaxUtils.check(beamBreakSensorOne.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorTwo.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorThree.enableLimitSwitch(false));
+    errors += SparkMaxUtils.check(beamBreakSensorFour.enableLimitSwitch(false));
+
+    return errors == 0;
+  }
+
+  public Command shoot() {
+    return new SequentialCommandGroup(
+      new InstantCommand(() -> frontMotor.set(ConveyerCal.PREPARE_TO_SHOOT_FRONT_SPEED)),
+      new InstantCommand(() -> backMotor.set(ConveyerCal.PREPARE_TO_SHOOT_BACK_SPEED)),
+      new InstantCommand(() -> currentNotePosition = ConveyerPositions.PARTIAL_NOTE),
+      new WaitCommand(ConveyerCal.NOTE_EXIT_TIME_SHOOTER_SECONDS),
+      stop(),
+      new InstantCommand(() -> currentNotePosition = ConveyerPositions.NO_NOTE),
+      new WaitUntilCommand(() -> backMotorEncoder.getVelocity() < ConveyerCal.MOTOR_VELOCITY_THRESHOLD_RPM),
+      new InstantCommand(() -> backMotorZeroPosition = backMotorEncoder.getPosition())
+    );
+  }
+
+  public Command scoreTrapOrAmp() {
+    return new SequentialCommandGroup(
+      new InstantCommand(() -> frontMotor.set(ConveyerCal.SCORE_AMP_TRAP_FRONT_SPEED)),
+      new InstantCommand(() -> backMotor.set(ConveyerCal.SCORE_AMP_TRAP_BACK_SPEED)),
+      new InstantCommand(() -> currentNotePosition = ConveyerPositions.PARTIAL_NOTE),
+      new WaitCommand(ConveyerCal.NOTE_EXIT_TIME_TRAP_AMP_SECONDS),
+      stop(),
+      new InstantCommand(() -> currentNotePosition = ConveyerPositions.NO_NOTE),
+      new WaitUntilCommand(() -> backMotorEncoder.getVelocity() < ConveyerCal.MOTOR_VELOCITY_THRESHOLD_RPM),
+      new InstantCommand(() -> backMotorZeroPosition = backMotorEncoder.getPosition())
+    );
+  }
+
+  public Command receive() {
+    return new SequentialCommandGroup(
+      new InstantCommand(() -> frontMotor.set(ConveyerCal.FRONT_RECEIVE_SPEED)),
+      new InstantCommand(() -> backMotor.set(0.0)),
+      new InstantCommand(() -> currentNotePosition = ConveyerPositions.PARTIAL_NOTE),
+      new WaitUntilCommand(() -> Math.abs(backMotorEncoder.getPosition() - backMotorZeroPosition) < ConveyerCal.NOTE_POSITION_THRESHOLD_DEGREES),
+      new InstantCommand(() -> backMotorZeroPosition = backMotorEncoder.getPosition()),
+      new InstantCommand(() -> frontMotor.set(ConveyerCal.BACK_OFF_POWER)),
+      new InstantCommand(() -> backMotor.set(ConveyerCal.BACK_OFF_POWER)),
+      new WaitUntilCommand(() -> Math.abs(backMotorEncoder.getPosition() - backMotorZeroPosition) < ConveyerCal.BACK_OFF_DEGREES),
```

Comment:
Maybe use front position for this? Back might slip on the note slightly but front def will not. Good idea tho